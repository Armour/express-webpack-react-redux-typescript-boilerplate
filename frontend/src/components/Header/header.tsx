import React from 'react';
import { Translation } from 'react-i18next';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Dropdown from 'components/Dropdown';
import { IGlobalStateRecord } from 'types/global';

const styles = require('./header.scss');

// Component

interface IHeaderStateProps {
  pathname: string;
}

export class Header extends React.Component<IHeaderStateProps> {
  constructor(props: IHeaderStateProps) {
    super(props);
  }

  public checkActive(urls: string[]) {
    let active = false;
    urls.forEach((url) => {
      if (this.props.pathname === '/' + url) {
        active = true;
      }
    });
    return active ? 'active' : '';
  }

  public componentDidMount() {
    const elems = document.querySelectorAll('.sidenav');
    M.Sidenav.init(elems);
  }

  public render() {
    const dropdownList = ['parallax'];
    return (
      <Translation ns='common'>
        {(t) => (
          <div>
            <nav>
              <div className='nav-wrapper'>
                <div className='container'>
                  <Link className='brand-logo' to='/'><img className={styles.logo} alt='logo' src={require('./assets/images/logo.png')}/></Link>
                  <a href='#' data-target='nav-mobile' className='sidenav-trigger'>
                    <i className='material-icons'>menu</i>
                  </a>
                  <ul id='nav-desktop' className='right hide-on-med-and-down'>
                    <li className={this.checkActive(['react'])} key='react'><Link to='/react'>{t('header.react')}</Link></li>
                    <li className={this.checkActive(dropdownList)} key='materialize'>
                      <a className='dropdown-button' href='#' data-target='header-dropdown-desktop'>{t('header.dropdown')}<i className='material-icons right'>arrow_drop_down</i></a>
                    </li>
                    <li className={this.checkActive(['404'])} key='404'><Link to='/404'>{t('header.notfound')}</Link></li>
                    <Dropdown id='header-dropdown-desktop' dropdownLists={dropdownList} />
                  </ul>
                </div>
              </div>
            </nav>
            <ul id='nav-mobile' className='sidenav'>
              <li className={this.checkActive(['react'])} key='react'><Link to='/react'>{t('header.react')}</Link></li>
              <li className={this.checkActive(dropdownList)} key='materialize'>
                <a className='dropdown-button' href='#' data-target='header-dropdown-mobile'>{t('header.dropdown')}<i className='material-icons right'>arrow_drop_down</i></a>
              </li>
              <li className={this.checkActive(['404'])} key='404'><Link to='/404'>{t('header.notfound')}</Link></li>
              <Dropdown id='header-dropdown-mobile' dropdownLists={dropdownList} />
            </ul>
          </div>
        )}
      </Translation>
    );
  }
}

// Container

const mapStateToProps = (state: IGlobalStateRecord): IHeaderStateProps => ({
  pathname: state.getIn(['router', 'location', 'pathname']),
});

export default connect(
  mapStateToProps,
  null,
)(Header);
