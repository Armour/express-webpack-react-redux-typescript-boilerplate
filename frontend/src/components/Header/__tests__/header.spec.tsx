import 'materialize-css';
import React, { Fragment } from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';

import { Header } from '../header';

jest.mock('components/Dropdown', () => 'Dropdown');

describe('Header', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <BrowserRouter>
        <Fragment>
          <Header pathname='/parallax' />,
        </Fragment>
      </BrowserRouter>,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
