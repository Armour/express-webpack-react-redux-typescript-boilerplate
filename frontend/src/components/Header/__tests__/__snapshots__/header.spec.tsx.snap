// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders correctly 1`] = `
Array [
  <div>
    <nav>
      <div
        className="nav-wrapper"
      >
        <div
          className="container"
        >
          <a
            className="brand-logo"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="logo"
              className="logo"
              src={
                Object {
                  "default": "test-file-stub",
                }
              }
            />
          </a>
          <a
            className="sidenav-trigger"
            data-target="nav-mobile"
            href="#"
          >
            <i
              className="material-icons"
            >
              menu
            </i>
          </a>
          <ul
            className="right hide-on-med-and-down"
            id="nav-desktop"
          >
            <li
              className=""
            >
              <a
                href="/react"
                onClick={[Function]}
              >
                header.react
              </a>
            </li>
            <li
              className="active"
            >
              <a
                className="dropdown-button"
                data-target="header-dropdown-desktop"
                href="#"
              >
                header.dropdown
                <i
                  className="material-icons right"
                >
                  arrow_drop_down
                </i>
              </a>
            </li>
            <li
              className=""
            >
              <a
                href="/404"
                onClick={[Function]}
              >
                header.notfound
              </a>
            </li>
            <Dropdown
              dropdownLists={
                Array [
                  "parallax",
                ]
              }
              id="header-dropdown-desktop"
            />
          </ul>
        </div>
      </div>
    </nav>
    <ul
      className="sidenav"
      id="nav-mobile"
    >
      <li
        className=""
      >
        <a
          href="/react"
          onClick={[Function]}
        >
          header.react
        </a>
      </li>
      <li
        className="active"
      >
        <a
          className="dropdown-button"
          data-target="header-dropdown-mobile"
          href="#"
        >
          header.dropdown
          <i
            className="material-icons right"
          >
            arrow_drop_down
          </i>
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="/404"
          onClick={[Function]}
        >
          header.notfound
        </a>
      </li>
      <Dropdown
        dropdownLists={
          Array [
            "parallax",
          ]
        }
        id="header-dropdown-mobile"
      />
    </ul>
  </div>,
  ",",
]
`;
